presto-iceberg/src/main/java/io/prestosql/plugin/iceberg/TypeConveter.java

```diff
@@ -0,0 +1,175 @@
+            .put(DoubleType.class, Types.DoubleType.get())
+            .put(BigintType.class, Types.LongType.get())
+            .put(RealType.class, Types.FloatType.get())
+            .put(IntegerType.class, Types.IntegerType.get())
+            .put(TimeType.class, Types.TimeType.get())
+            .put(TimestampType.class, Types.TimestampType.withoutZone())
+            .put(TimestampWithTimeZoneType.class, Types.TimestampType.withZone())
+            .put(VarcharType.class, Types.StringType.get())
+            .build();
+
+    private TypeConveter() {}
+
+    public static Type toPrestoType(org.apache.iceberg.types.Type type, TypeManager typeManager)
+    {
+        switch (type.typeId()) {
+            case BOOLEAN:
+                return BooleanType.BOOLEAN;
+            case BINARY:
+            case FIXED:
+                return VarbinaryType.VARBINARY;
+            case DATE:
+                return DateType.DATE;
+            case DECIMAL:
+                Types.DecimalType decimalType = (Types.DecimalType) type;
+                return DecimalType.createDecimalType(decimalType.precision(), decimalType.scale());
+            case DOUBLE:
+                return DoubleType.DOUBLE;
+            case LONG:
+                return BigintType.BIGINT;
+            case FLOAT:
+                return RealType.REAL;
+            case INTEGER:
+                return IntegerType.INTEGER;
+            case TIME:
+                return TimeType.TIME;
+            case TIMESTAMP:
+                Types.TimestampType timestampType = (Types.TimestampType) type;
+                if (timestampType.shouldAdjustToUTC()) {
+                    return TimestampWithTimeZoneType.TIMESTAMP_WITH_TIME_ZONE;
+                }
+                return TimestampType.TIMESTAMP;
+            case UUID:
+            case STRING:
+                return VarcharType.createUnboundedVarcharType();
+            case LIST:
+                Types.ListType listType = (Types.ListType) type;
+                return new ArrayType(toPrestoType(listType.elementType(), typeManager));
+            case MAP:
+                Types.MapType mapType = (Types.MapType) type;
+                Type keyType = toPrestoType(mapType.keyType(), typeManager);
+                Type valueType = toPrestoType(mapType.valueType(), typeManager);
+                MethodHandle keyNativeEquals = typeManager.resolveOperator(OperatorType.EQUAL, ImmutableList.of(keyType, keyType));
+                MethodHandle keyBlockNativeEquals = compose(keyNativeEquals, nativeValueGetter(keyType));
+                MethodHandle keyBlockEquals = compose(keyNativeEquals, nativeValueGetter(keyType), nativeValueGetter(keyType));
+                MethodHandle keyNativeHashCode = typeManager.resolveOperator(OperatorType.HASH_CODE, ImmutableList.of(keyType));
+                MethodHandle keyBlockHashCode = compose(keyNativeHashCode, nativeValueGetter(keyType));
+                return new MapType(
```

Comment:
Why not just resolve the map type from the TypeManager?

Reply:
Thanks, I missed this one